cmd/cut.lisp

```diff
@@ -0,0 +1,92 @@
+(import "lib/options/options.inc")
+
+(defq usage `(
+(("-h" "--help")
+"Usage: cut [options] [file ...]
+
+	options:
+		-h --help: this help info.
+		-f --fields list: select only these fields (comma-separated).
+		-d --delimiter char: use CHAR as field delimiter (default TAB).
+		-c --characters list: select only these character positions.
+
+	Extract fields or characters from lines of input.
+	LIST format: N, N-M, N-, -M (comma separated).
+
+	Examples:
+		cut -f 1,3 file.txt     # fields 1 and 3
+		cut -f 2- file.txt      # field 2 to end
+		cut -c 1-10 file.txt    # characters 1-10")
+(("-f" "--fields") ,(opt-str 'opt_fields))
+(("-d" "--delimiter") ,(opt-str 'opt_delim))
+(("-c" "--characters") ,(opt-str 'opt_chars))
+))
+
+(defun parse-ranges (spec)
+	;parse range specification like "1,3-5,7-"
+	(defq ranges (list))
+	(each (lambda (part)
+			(cond
+				;range N-M
+				((defq pos (find "-" part))
+					(defq start (if (> pos 0) (str-to-num (slice part 0 pos)) 1))
+					(defq end (if (< pos (dec (length part))) (str-to-num (slice part (inc pos) -1)) 999999))
+					(push ranges (list start end)))
+				;single number N
+				:t
+					(defq n (str-to-num part))
+					(push ranges (list n n))))
+		(split spec ","))
+	ranges)
+
+(defun in-ranges (n ranges)
+	;check if number n is in any of the ranges
+	(some (lambda (r) (and (>= n (first r)) (<= n (second r)))) ranges))

```

Comment:
Lisp can do '(<= r1 n r2)' in a single test ! Numeric compares like this are multi argument ! This applies to all of them.